lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -781,25 +781,27 @@ namespace Js
         }
         bool result = true;
         confirmedReady = true;
-        EnsureChildModuleSet(GetScriptContext());
-        childrenModuleSet->EachValue([&](SourceTextModuleRecord* childModuleRecord) {
-            if (childModuleRecord->ParentsNotified())
-            {
-                return false;
-            }
-            else
-            {
-                if (childModuleRecord->ConfirmChildrenParsed())
+        if (childrenModuleSet != nullptr)
```

Comment:
Drive by improvement - previously in the case of no child modules an empty array was allocated here.